ee/audit/run_ee.go

```diff
@@ -107,37 +107,161 @@ func run() error {
-	x.Check2(file.ReadAt(t, iterator))
-	iterator = iterator + int64(len(x.VerificationText))
+		ct := make([]byte, len(x.VerificationText))
+		_, err = file.ReadAt(ct, iterator)
+		if err != nil {
+			return nil, 0, err
+		}
+		iterator = iterator + int64(len(x.VerificationText))
 
-	stream := cipher.NewCTR(block, iv)
-	stream.XORKeyStream(t, t)
-	if string(t) != x.VerificationText {
-		return errors.New("invalid encryption key provided. Please check your encryption key")
+		text := make([]byte, len(x.VerificationText))
+		stream := cipher.NewCTR(block, iv)
+		stream.XORKeyStream(text, ct)
+		if string(text) != x.VerificationText {
+			return nil, 0, errors.New("invalid encryption key provided. Please check your encryption key")
+		}
+		return iv, iterator, nil
 	}
 
-	for {
-		// if its the end of data. finish decrypting
-		if iterator >= stat.Size() {
-			break
+	// [12]byte baseIV + [4]byte len(x.VerificationTextDeprecated) + [11]byte x.VerificationTextDeprecated
+	decryptHeaderDeprecated := func() ([]byte, int64, error) {
+		var iterator int64 = 0
+
+		iv := make([]byte, aes.BlockSize)
+		_, err := file.ReadAt(iv, iterator)
+		if err != nil {
+			return nil, 0, err
 		}
-		x.Check2(file.ReadAt(iv[12:], iterator))
-		iterator = iterator + 4
+		iterator = iterator + aes.BlockSize
 
-		content := make([]byte, binary.BigEndian.Uint32(iv[12:]))
-		x.Check2(file.ReadAt(content, iterator))
-		iterator = iterator + int64(binary.BigEndian.Uint32(iv[12:]))
+		ct := make([]byte, len(x.VerificationTextDeprecated))
+		_, err = file.ReadAt(ct, iterator)
+		if err != nil {
+			return nil, 0, err
+		}
+		iterator = iterator + int64(len(x.VerificationTextDeprecated))
+
+		text := make([]byte, len(x.VerificationTextDeprecated))
 		stream := cipher.NewCTR(block, iv)
-		stream.XORKeyStream(content, content)
-		x.Check2(outfile.Write(content))
+		stream.XORKeyStream(text, ct)
+		if string(text) != x.VerificationTextDeprecated {
+			return nil, 0, errors.New("invalid encryption key provided. Please check your encryption key")
+		}
+		return iv, iterator, nil
 	}
+
+	useDeprecated := false
+	iv, iterator, err := decryptHeader()
+	if err != nil {
+		// might have an old audit log
+		iv2, iterator2, err := decryptHeaderDeprecated()
+		if err != nil {
+			return errors.New("invalid encryption key provided. Please check your encryption key")
+		}
+		// found old audit log
+		useDeprecated = true
+		iv, iterator = iv2, iterator2
+	}
+
+	var count int
+
+	// encrypted writes each have the form below
+	// IV generated for each write
+	// #################################################################
+	// #####   [16]byte IV + [4]byte uint32(len(p)) + [:]byte p    #####
+	// #################################################################
+	decryptBody := func() {
+		for {
+			count++
+			// if its the end of data. finish decrypting
+			if iterator >= stat.Size() {
+				break
+			}
+			n, err := file.ReadAt(iv, iterator)
+			if err != nil {
```

Comment:
shouldn't we break the loop or something?

Reply:
yes, added break statement here